### DISCUSSION TITLE
How to get the road layer 'Ground Truth' directly from the Carla map

### DISCUSSION THREAD

**Discussion by tbrayer2** [carla-simulator/carla#8420] [Q&A]
Hello, 

I work on a framework which produces a custom bird eye view segmentation map of the scene. I’m interested in the following labels: road, lane dividers, crosswalk areas, and stop line polygons. At first, I attached a facing down segmentation sensor to the ego vehicle and grabbed its output. This did not yield what I desired, as shadows of buildings and vegetation on the road excluded pixels from the road mask. In addition, I would like to get the mask of the lane divider and crosswalk regardless of where the dash lines appear. I just want the mask of all the pixels which are on the lane divider from a certain viewpoint.  
Therefore, I would like to get the locations of the road, lane dividers, crosswalk area, and stop line area directly from the map of the world. The carla.map API only produces waypoints which are centered at center of the lane, therefore is not useful for extracting the masks for my desired labels. Can you please recommend a way to get the locations of the points belonging to the classes I mentioned (road, lane divider, crosswalk area and stop line area)?

**Comment by reynerliu** [comment]
@tbrayer2 Hello? Have you solved this question, I have the as same target as you. If you can provide me some ideas I would be very appreciated!

**Reply by tbrayer2** [reply]
Yes, I load and parse the XODR file of the town, and I extract the geometric representation from the geometry elements and the lane section elements. I think that there might be also available packages for parsing the XODR files.
